tests/Doctrine/Tests/Common/Cache/MemcachedCacheTest.php

```diff
@@ -49,6 +49,33 @@ public function testGetMemcachedReturnsInstanceOfMemcached()
         $this->assertInstanceOf('Memcached', $this->_getCacheDriver()->getMemcached());
     }
 
+    public function testDoContains()
+    {
+        $driver = $this->_getCacheDriver();
+        $reflection = new \ReflectionClass($driver);
+        $method = $reflection->getMethod('doContains');
+        $method->setAccessible(true);
+
+        $testKey = __CLASS__.'#'.__METHOD__;
+
+        $this->memcached->set($testKey, false);
+
+        $this->assertTrue($method->invokeArgs($driver, [$testKey]));
+        $this->assertFalse($method->invokeArgs($driver, [$testKey.'2']));
+
+        $memcached = new Memcached();
+        $memcached->addServer('0.0.0.1', 11211); // fake server is not available
```

Comment:
you should write 2 separate tests (one covering the storage of ``false`` and another one covering the handling of errors with a fake server) instead of mixing both in the same test. We need them to be able to fail independently. Thus, it would give more info about what each test is covering (as they would each have a descriptive name)